noah/src/test/scala/edu/uci/ics/cloudberry/noah/kafka/TestProducer.scala

```diff
@@ -0,0 +1,76 @@
+package edu.uci.ics.cloudberry.noah.kafka
+
+import java.util.concurrent.{BlockingQueue, LinkedBlockingDeque}
+
+import com.twitter.hbc.core.Client
+import edu.uci.ics.cloudberry.noah.GeneralProducerKafka
+import edu.uci.ics.cloudberry.noah.feed.Config
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.mockito._
+import org.specs2.mock.Mockito
+import org.specs2.mutable.Specification
+import org.mockito.Mockito._
+import com.twitter.hbc.core.endpoint.Location
+
+class TestProducer extends Specification with Mockito {
+
+  "FileProducer" should {
+    "read from a .gz file and send records to kafka" in {
+      val generalProducerKafka = mock[GeneralProducerKafka]
+      val kafkaProducer = mock[KafkaProducer[String, String]]
+
+      val mockConfig = mock[Config]
+      val fileProducer = new FileProducer
+      when (mockConfig.getFilePath).thenReturn("/Tweet")
+      when (mockConfig.getKfkTopic).thenReturn("")
+      fileProducer.run(mockConfig, generalProducerKafka, kafkaProducer)
+
+      val argument = ArgumentCaptor.forClass(classOf[String])
+      Mockito.verify(generalProducerKafka, times(3)).store(any, argument.capture(), any)
+
+      val result = argument.getAllValues
+      result should have size 3
+      result.get(0) must_== "test1"
+      result.get(1) must_== "test2"
+      result.get(2) must_== "test3"
+    }
+  }
+
+  "TweetsProducer" should {
+    val tweetsProducer = new TweetsProducer
+    val mockConfig = mock[Config]
+    when (mockConfig.getTrackTerms).thenReturn(Array("trump"))
+
+    val loc: Array[Location] = new Array(0)
+    when (mockConfig.getTrackLocation).thenReturn(loc)
+    when (mockConfig.getConsumerKey).thenReturn("ck")
+    when (mockConfig.getConsumerSecret).thenReturn("cs")
+    when (mockConfig.getToken).thenReturn("tk")
+    when (mockConfig.getTokenSecret).thenReturn("ts")
+
+    /**"create a Twitter Client and set up connection" in {
```

Comment:
oh, what I meant is to add a comment to this test to explain why we skip it for now .